rust/ql/lib/codeql/rust/frameworks/stdlib/Env.qll

```diff
@@ -0,0 +1,33 @@
+/**
+ * Provides modeling for the `std::env` library.
+ */
+
+private import rust
+private import codeql.rust.Concepts
+
+/**
+ * A call to `std::env::args` or `std::env::args_os`.
+ */
+private class StdEnvArgs extends CommandLineArgsSource::Range {
+  StdEnvArgs() {
+    this.asExpr().getExpr().(CallExpr).getExpr().(PathExpr).getPath().getResolvedPath() = ["crate::env::args", "crate::env::args_os"]
```

Comment:
@redsun82 these paths don't look right - I assume the correct path would be something like `std::env::args` or `crate::std::env::args`?  They do match both what we get in the tests and what we see in real databases at the moment.

Reply:
(fixing this will probably be follow-up work, what we have works well enough right now and is needed to get queries working)

Reply:
hmm, this might have to do with the prelude, I'll have a look

Reply:
ah, no, currently that is expected.

The thing is that a path only makes sense in the context of a crate (there is no such thing as a global concept of path across crates). So that path is the correct one _relative to the std crate_.

The other piece of information is the (highly non-standard and prone to be changed) `getCrateOrigin`, which gives `lang:std` for the std lib.

we could theoretically combine the two, it's just that while I'm quite confident of the correctness of the resolved path (up to the fact that it may not cover all entities yet), I'm still unsure about the effectiveness of `getCrateOrigin`.

Reply:
OK.  It sounds like we could check `getCrateOrigin` for models where we're worried about ambiguity (`"crate::get"` makes me nervous), but perhaps not check it for cases where we aren't (e.g. `"crate::env::vars"`) since that will be more robust if `getCrateOrigin` is potentially prone to change.  Alternatively we could fall back on other details such as types to ensure our models are matching the right things - to the extent that we extract enough information.

Reply:
I've push a commit adding the restriction I described above.

I've also added an agenda item next week to discuss crate origins, how we can get reliable information and how they will interact with models-as-data.